lib/rules/valid-jsdoc.js

```diff
@@ -226,8 +226,11 @@ module.exports = {
         function checkJSDoc(node) {
             const jsdocNode = sourceCode.getJSDocComment(node),
                 functionData = fns.pop(),
-                params = Object.create(null);
+                params = Object.create(null),
+                hasParamsTags = [];
             let hasReturns = false,
+                hasReturnsTag,
+                hasParams = false,
```

Comment:
I think the `has` prefix here is intended to refer to boolean conditions (e.g. the `hasReturns` variable describes the boolean condition of whether the JSDoc comment has any returns). You've created some other variables prefixed with `has`, but they don't seem to be describing boolean conditions, which was confusing to me as I was reading through the code. I would recommend removing the `has` prefix for those variables (e.g. renaming `hasReturnsTag` to `returnsTag`).